scipy-stubs/signal/_peak_finding.pyi

```diff
@@ -1,37 +1,72 @@
+    right_thresholds: NotRequired[npt.NDArray[np.float64]]
+    prominences: NotRequired[npt.NDArray[np.float64]]
+    left_bases: NotRequired[npt.NDArray[np.intp]]
+    right_bases: NotRequired[npt.NDArray[np.intp]]
+    width_heights: NotRequired[npt.NDArray[np.float64]]
+    left_ips: NotRequired[npt.NDArray[np.float64]]
+    right_ips: NotRequired[npt.NDArray[np.float64]]
+    plateau_sizes: NotRequired[npt.NDArray[np.intp]]
+    left_edges: NotRequired[npt.NDArray[np.intp]]
+    right_edges: NotRequired[npt.NDArray[np.intp]]
+
+def argrelmin(
+    data: npt.NDArray[np.generic], axis: int = 0, order: int = 1, mode: _Mode = "clip"
+) -> tuple[npt.NDArray[np.intp], ...]: ...
+def argrelmax(
+    data: npt.NDArray[np.generic], axis: int = 0, order: int = 1, mode: _Mode = "clip"
+) -> tuple[npt.NDArray[np.intp], ...]: ...
+def argrelextrema(
+    data: npt.NDArray[np.generic],
+    comparator: Callable[[npt.NDArray[np.generic], npt.NDArray[np.generic]], npt.NDArray[np.bool_]],
+    axis: int = 0,
+    order: int = 1,
+    mode: _Mode = "clip",
+) -> tuple[npt.NDArray[np.intp], ...]: ...
+def peak_prominences(x: npt.ArrayLike, peaks: npt.ArrayLike, wlen: int | None = None) -> _ProminencesResult: ...
 def peak_widths(
-    x: Untyped,
-    peaks: Untyped,
+    x: npt.ArrayLike,
+    peaks: npt.ArrayLike,
     rel_height: float = 0.5,
-    prominence_data: Untyped | None = None,
-    wlen: Untyped | None = None,
-) -> Untyped: ...
+    prominence_data: _ProminencesResult | None = None,
+    wlen: int | None = None,
+) -> _WidthsResult: ...
 def find_peaks(
-    x: Untyped,
-    height: Untyped | None = None,
-    threshold: Untyped | None = None,
-    distance: Untyped | None = None,
-    prominence: Untyped | None = None,
-    width: Untyped | None = None,
-    wlen: Untyped | None = None,
+    x: npt.ArrayLike,
+    height: float | npt.NDArray[np.float64] | tuple[float | None, float | None] | None = None,
+    threshold: float | npt.NDArray[np.float64] | tuple[float | None, float | None] | None = None,
+    distance: np.float64 | None = None,
+    prominence: float | npt.NDArray[np.float64] | tuple[float | None, float | None] | None = None,
+    width: float | npt.NDArray[np.float64] | tuple[float | None, float | None] | None = None,
+    wlen: int | None = None,
     rel_height: float = 0.5,
-    plateau_size: Untyped | None = None,
-) -> Untyped: ...
+    plateau_size: int | npt.NDArray[np.intp] | tuple[int | None, int | None] | None = None,
+) -> tuple[npt.NDArray[np.intp], _FindPeaksResultsDict]: ...
 def find_peaks_cwt(
-    vector: Untyped,
-    widths: Untyped,
-    wavelet: Untyped | None = None,
-    max_distances: Untyped | None = None,
-    gap_thresh: Untyped | None = None,
-    min_length: Untyped | None = None,
+    vector: npt.ArrayLike,
+    widths: npt.ArrayLike,
+    wavelet: Callable[Concatenate[int, float, ...], npt.NDArray[np.float64]] | None = None,
+    max_distances: Sequence[int] | None = None,
```

Comment:
I know it's a bit weird, but an `ndarray` isn't assignable to a `Sequence`, which the docs say that this should accept

Reply:
Oh thats a bit weird/annoying. The reason why I chose `Sequence` is because I didn't look at the docs for `find_peaks_cwt` (because it is a bit long) but instead for the function that `max_distances` gets passed to which is called `_identify_ridge_lines` which has says that `max_distances` is a 1-D sequence.

The source code for that function only calls `len` and indexes `max_distances` which fits the definition of `Sequence`. The values from `max_distances` gets compared to `np.intp` as well.

Would the type then be like `onpt.Array[tuple[int], AnyInt]`? That might still be too narrow considering that `Sequence` is a relatively weak type constraint.

Reply:
The problem with `Sequence` is that apart from `__getitem__` it also has several other methods like `.count()` and `.index()` that aren't present in `np.ndarray` 🤷🏻 

And yea I agree that `onpt.Array` (or some other direct `np.ndarray` alias) would be indeed too narrow. So you could use e.g. `_ArrayLikeInt_co`, which is compatible with both `Sequence[int]` and `npt.NDArray[integer[Any]]`. And as far as I'm concerned, the fact that it would also allow integer scalars isn't much of a problem, as we're talking about the input of a function.